### PR TITLE
#5

### PR DESCRIPTION
**Pull request by renrenovich**

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,47 +0,0 @@
-asgiref==3.5.2
```
Comment (Yakov-Varnaev) [inline requirements.txt:1]: And why file is cleaned?

**Review by Yakov-Varnaev — CHANGES_REQUESTED** [review]

File: inventory/services.py [hunk 2]
```diff
@@ -17,5 +18,18 @@ class MaterialCreateService(ModelCreateService, BaseMaterialService):
     model = Material
 
 
-class MaterialDestroyService(ModelDestroyService):
-    model = Material
+class MaterialDestroyService:
+    def __init__(self, instance):
```
Comment (max-bazarov) [inline inventory/services.py:22]: Try to copy Yakov's solution.
self.instance = instance
self.model = instance.__class__
etc.

File: inventory/tests.py [hunk 3]
```diff
@@ -35,3 +38,32 @@ def setUpClass(cls):
         }
 
         cls.instance = Material.objects.create(**cls.data)
+
+
+class TestMaterialDestroyService(APITestCase):
```
Comment (max-bazarov) [inline inventory/tests.py:43]: Inherit from BaseCRUDViewTest also

**Review by max-bazarov — CHANGES_REQUESTED** [review]

File: inventory/tests.py (hunk 3 above)
Comment (Yakov-Varnaev) [reply inventory/tests.py:43]: Not BaseCRUDViewTest, but BaseCRUDArchiveViewTest.

File: employees/migrations/0001_employees_models.py [hunk 4]
```diff
@@ -40,4 +40,4 @@ class Migration(migrations.Migration):
                 'db_table': 'master_procedure',
             },
         ),
-    ]
+    ]
```
Comment (Yakov-Varnaev) [inline employees/migrations/0001_employees_models.py:43]: add new line in the end file to remove this file from pr.

File: inventory/serializers.py [hunk 5]
```diff
@@ -0,0 +1,10 @@
+from rest_framework import serializers
+
+from inventory.models import Material
+
+
+class MaterialSerializer(serializers.ModelSerializer):
+    """Сериализатор для материала"""
```
Comment (Yakov-Varnaev) [inline inventory/serializers.py:7]: Use English

File: inventory/services.py [hunk 6]
```diff
@@ -17,5 +18,18 @@ class MaterialCreateService(ModelCreateService, BaseMaterialService):
...
+
+    def has_related(self) -> bool:
+        return self.instance.uses.exists()
+
+    def destroy(self):
+        if self.has_related():
+            raise Exception(
+                f'Material cannot be deleted, because it is used material'
+            )
+        self.instance.delete()
```
Comment (Yakov-Varnaev) [inline inventory/services.py:33]: You should return id of deleted instance.

File: inventory/tests.py [hunk 7]
```diff
@@ -6,15 +6,18 @@
...
 from core.tests import (BaseArchiveServiceTest, BaseCreateServiceTest,
                         BaseDestroyServiceTest)
+from employees.models import Employee, MasterProcedure
+from procedures.models import Procedure
+from purchases.models import UsedMaterial, Purchase, PurchaseProcedure
 
 from .models import Material, MaterialUnits
 from .services import MaterialCreateService, MaterialDestroyService
-
+from rest_framework.test import APITestCase
```
Comment (Yakov-Varnaev) [inline inventory/tests.py:15]: use isort to fix imports

File: inventory/tests.py [hunk 8]
```diff
@@ -35,3 +38,32 @@ def setUpClass(cls):
...
+        self.instance = None
+
+    data = {
+        'name': 'Hair Color',
+        'price': Decimal('1.11'),
+        'unit': MaterialUnits.GRAMMS.value
+    }
+
+    @classmethod
+    def SetUpClass(cls):
```
Comment (Yakov-Varnaev) [inline inventory/tests.py:56]: There is no such method. It's `setUpClass`.

Comment (Yakov-Varnaev) [reply inventory/tests.py:56]: And you should call super method too.

File: inventory/views.py [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+from inventory.models import Material
+from inventory.serializers import MaterialSerializer
+from inventory.services import MaterialDestroyService
+
+
+class MaterialViewSet(ModelViewSet):
+    queryset = Material.objects.filter(archived=False)
+    serializer_class = MaterialSerializer
+
+    def update(self, request, *args, **kwargs):
```
Comment (Yakov-Varnaev) [inline inventory/views.py:15]: partial_update also should be redefined

**Review by Yakov-Varnaev — CHANGES_REQUESTED** [review]

File: inventory/serializers.py (hunk 5 above)
Comment (max-bazarov) [reply inventory/serializers.py:7]: @Yakov-Varnaev Is there any sense to leave such comments? This code is self-documented. Everybody understands that it is serializer for materials model

Comment (Yakov-Varnaev) [reply inventory/serializers.py:7]: Agreed

File: inventory/tests.py (hunk 3 above)
Comment (max-bazarov) [reply inventory/tests.py:43]: It would be better to rename this Test to TestMaterialDestroyViews and relocate testing destroy service to test above

**Review by max-bazarov — CHANGES_REQUESTED** [review]

**Review by max-bazarov — APPROVED** [review]

**Review by Yakov-Varnaev — APPROVED** [review]

**Pull request merged**